platform/ext_bitbucket.go

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2021-2024 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package platform
+
+import (
+	"context"
+	"fmt"
+	"github.com/JetBrains/qodana-cli/v2024/sarif"
+	bbapi "github.com/reviewdog/go-bitbucket" // adapted from https://raw.githubusercontent.com/reviewdog/reviewdog/master/LICENSE
+	log "github.com/sirupsen/logrus"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+)
+
+const (
+	httpTimeout              = time.Second * 10
+	bitBucketAnnotationLimit = 1000
+	bitBucketReporter        = "JetBrains Qodana"
+	bitBucketAvatar          = "https://avatars.githubusercontent.com/u/139879315"
+	bitBucketReportFailed    = "FAILED"
+	bitBucketReportPassed    = "PASSED"
+	bitBucketReportType      = "BUG"
+	bitBucketAnnotationType  = "CODE_SMELL"
+
+	// https://developer.atlassian.com/cloud/bitbucket/rest/api-group-reports/#api-repositories-workspace-repo-slug-commit-commit-reports-reportid-annotations-annotationid-put-request
+	bitBucketHigh   = "HIGH"
+	bitBucketMedium = "MEDIUM"
+	bitBucketLow    = "LOW"
+	bitBucketInfo   = "INFO"
+
+	pipelineProxyURL = "http://localhost:29418"
+	pipeProxyURL     = "http://host.docker.internal:29418"
+)
+
+var (
+	toBitBucketSeverity = map[string]string{
+		sarifError:     bitBucketHigh,
+		sarifWarning:   bitBucketMedium,
+		sarifNote:      bitBucketLow,
+		qodanaCritical: bitBucketHigh,
+		qodanaHigh:     bitBucketMedium,
```

Comment:
Naively I would map high to high here - is there a reason for medium? If yes, let's comment that so that our future selves don't treat it as a bug ;)

Reply:
Yeah, HIGH is better here, will update